alpha/books/RayTracingInOneWeekend.html

```diff
@@ -2684,9 +2701,10 @@
 A Data Structure to Describe Ray-Object Intersections
 ------------------------------------------------------
 The `hit_record` is to avoid a bunch of arguments so we can stuff whatever info we want in there.
-You can use arguments instead; it’s a matter of taste. Hittables and materials need to know each
-other so there is some circularity of the references. In C++ you just need to alert the compiler
-that the pointer is to a class, which the “class material” in the hittable class below does:
+You can use arguments instead of an encapsulated type, it’s just a matter of taste. Hittables and
+materials need to be able to reference the other's type in code so there is some circularity of the
+references. In C++ you just need to alert the compiler that the pointer is to a class, which the
+`class material;` in the hittable class below does:
```

Comment:
Hmmm, I think this could be improved. What do you think about this?

> In C++ we add the line `class material;` to tell the compiler that `material` is a class that will be defined later.

Optionally, we could continue with

> Since we're just going to specify a pointer to the class, the compiler doesn't need to know the details of the class, solving the circular reference issue.